internal/build/imgsrc/ensure_builder.go

```diff
@@ -159,9 +201,30 @@ const (
 	InvalidMachineCount
 	BuilderMachineNotStarted
 	ShouldReplaceBuilderMachine
+
+	buildkitGRPCPort = 1234
 )
 
-func validateBuilder(ctx context.Context, app *fly.App) (*fly.Machine, error) {
+// validateBuilder returns a machine if it is availabe for building images.
+func (p *Provisioner) validateBuilder(ctx context.Context, app *fly.App) (*fly.Machine, error) {
+	machine, err := p.validateBuilderMachine(ctx, app)
+	if err != nil {
+		return nil, err
```

Comment:
because there is an existing `return machine, BuilderMachineNotStarted` case in `validateBuilderMachine` below, this needs to return the machine otherwise it causes a nil pointer deference panic.

```suggestion
		return machine, err
```

Reply:
Oh why. Fixed.